### DISCUSSION TITLE
parentGrainReconstructor

### DISCUSSION THREAD

**Discussion by ghost** [mtex-toolbox/mtex#666] [Ask Anything] [answered]
Hi Ralf, 

In the step compute parent orientations, the parent phase was not reconstructed using the job.calcParentFromGraph function.

I don't have the FCC parent phase in my data, conssequently, in my CSList. So, I'm having trouble in the initial guess of orientation relationship. 

Please, how can I deal with this problem?

**Comment by ralfHielscher** [comment]
Hi Ariane,
in case you are asking how to use `parentGrainReconstructor` - you have to download [MTEX 5.6.beta](https://github.com/mtex-toolbox/mtex/releases/download/mtex-5.6.beta.2/mtex-5.6.beta.2.zip).

A happy new year,

Ralf.

**Reply by ghost** [reply]
Hi Ralf,

In the step compute parent orientations, the parent phase was not reconstructed using the job.calcParentFromGraph function.

I don't have the FCC parent phase in my data, conssequently, in my CSList. So, I'm having trouble in the initial guess of orientation relationship.

Please, how can I deal with this problem?

**Comment by ralfHielscher** [comment, marked as answer]
The documentation says

> The parentGrainReconstructor guesses from the EBSD data what is the parent and what is the child phase. If this guess is not correct it might be specified explicitely by defining an initial guess for the parent to child orientation relationship first and passing it as a third argument to parentGrainReconstructor.

This means you need to define the parent phase yourself by
```matlab
csParent = crystalSymmetry('432','mineral','FCC') % whatever your parent phase is
p2c = orientation.KurdjumovSachs(csParent, ebsd('childPhase').CS)
job = parentGrainReconstructor(ebsd,grains,p2c)
```
Ralf.

**Reply by ghost** [reply]
Ralf, 
I tried it before. The parent grains were not reconstructed. 
```
job.calcParentFromGraph
computing parent grain orientations: 100%
 
ans = parentGrainReconstructor (show methods, plot)
 
 phase   mineral       symmetry  grains  area   reconstructed
 parent  Iron (Gamma)  "432"     17596   0%     0%           
 child   Iron (Alpha)  "432"     660     0.49%               
 
 parent to child OR: (340.1°,9°,63.6°)
   fit with child to child misorientations: 4.9°, 9.2°, 14°, 17° (quintiles)
   closest ideal OR: (31-2) || (12-1)   [112] || [012]   error: 2°
```

**Comment by ralfHielscher** [comment]
The reason is that you child 2 child misorientations do not fit the parent
to child orientation relationship.

what is the output of `job.calcParent2child`

Ralf.

**Comment by ralfHielscher** [comment]
it seems you mixed the parent and the child phase. What es the output of
```matlab
csParent = crystalSymmetry('432','mineral','FCC') % whatever your parent phase is
p2c = orientation.KurdjumovSachs(csParent, ebsd('childPhase').CS)
job = parentGrainReconstructor(ebsd,grains,p2c)
```
Ralf

**Reply by ghost** [reply]
Half,
I agree with you that parent and the child phases were mixed.

` csChild = ebsd('Iron (Alpha)').CS
csParent = crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Iron (Gamma)', 'color', [0.56 0.74 0.56])

p2c = orientation.KurdjumovSachs(csParent,csChild)

job = parentGrainReconstructor(ebsd,grains,p2c) `

job = parentGrainReconstructor (show methods, plot)
 
 phase   mineral       symmetry  grains  area  reconstructed
 parent  Iron (Gamma)  "432"     0       0%    0%           
 child   Iron (Alpha)  "432"     18256   100%               
 
 parent to child OR: (335.8°,10.5°,65.8°)
   fit with child to child misorientations: 3°, 4°, 4.8°, 5.6° (quintiles)
   closest ideal OR: (10-1) || (11-1)   [111] || [011]   error: 0°

`job.calcParent2Child`

ans = parentGrainReconstructor (show methods, plot)
 
 phase   mineral       symmetry  grains  area  reconstructed
 parent  Iron (Gamma)  "432"     0       0%    0%           
 child   Iron (Alpha)  "432"     18256   100%               
 
 parent to child OR: (340.1°,9°,63.6°)
   fit with child to child misorientations: 1.6°, 2.3°, 2.9°, 3.9° (quintiles)
   closest ideal OR: (31-2) || (12-1)   [112] || [012]   error: 2°

I did the intermediate steps according to documentation.

`job.clusterGraph('inflationPower',1.6)`

ans = parentGrainReconstructor (show methods, plot)
 
 phase   mineral       symmetry  grains  area  reconstructed
 parent  Iron (Gamma)  "432"     0       0%    0%           
 child   Iron (Alpha)  "432"     18256   100%               
 
 parent to child OR: (340.1°,9°,63.6°)
   fit with child to child misorientations: 1.6°, 2.3°, 2.9°, 3.9° (quintiles)
   closest ideal OR: (31-2) || (12-1)   [112] || [012]   error: 2°
 
 graph: 17596 grains in 600 clusters + 673 single grain clusters

`job.calcParentFromGraph`

computing parent grain orientations: 100%
 
ans = parentGrainReconstructor (show methods, plot)
 
 phase   mineral       symmetry  grains  area   reconstructed
 parent  Iron (Gamma)  "432"     17596   0%     0%           
 child   Iron (Alpha)  "432"     660     0.49%               
 
 parent to child OR: (340.1°,9°,63.6°)
   fit with child to child misorientations: 4.9°, 9.2°, 14°, 17° (quintiles)
   closest ideal OR: (31-2) || (12-1)   [112] || [012]   error: 2°

Please, did I miss a step?

Is there any chance that parentPhaseId has not been identified correctly?
I don't have this problem when the data has an FCC phase (Pixels = 0). I believe that the phases are not mixed when the phaseId is extracted from ebsd.CSList. But, I don't know if I understood the code correctly.

Sorry if I couldn't explain the problem clearly.

**Reply by ralfHielscher** [reply]
Hi Arianne,

there is indeed a bug when the parent phase was not part of the EBSD data. I fixed it here 696ca1e. Are you able to apply this patch? You need to change only 3 lines of code within MTEX.

Except from this. Your grains have already been reconstructed. You can access them by `job.grains('Iron (Gamma)')`

Ralf.

**Reply by ghost** [reply]
Ralf,
I changed the code. Thank you!
